### PR TITLE
New component structure

### PR DESCRIPTION
**Pull request by MiroslavPetrik**
fix https://github.com/Kamahl19/react-starter/issues/38

fix https://github.com/Kamahl19/react-starter/issues/17

### CODE REVIEW

File: src/features/auth/AuthApp.js [hunk 1]
```diff
@@ -6,31 +6,27 @@ import IsLoggedIn from '../../common/services/user/guards/IsLoggedIn';
 import IsAnonymous from '../../common/services/user/guards/IsAnonymous';
 import LoginGuard from '../../common/services/user/guards/LoginGuard';
 
-import ForgottenPasswordContainer from './containers/ForgottenPasswordContainer';
-import LoginContainer from './containers/LoginContainer';
-import LogoutContainer from './containers/LogoutContainer';
-import ResetPasswordContainer from './containers/ResetPasswordContainer';
-import SignUpContainer from './containers/SignUpContainer';
 import Header from './components/Header';
+import ForgottenPassword from './components/ForgottenPassword/';
```
Comment (Kamahl19) [inline src/features/auth/AuthApp.js]: why tailing slash?

File: src/features/auth/components/Logout/index.js [hunk 2]
```diff
@@ -20,7 +18,7 @@ class LogoutContainer extends Component {
   }
 
   render() {
-    return <AuthSpinner />;
+    return <h1>Logging you out...</h1>;
```
Comment (Kamahl19) [inline src/features/auth/components/Logout/index.js]: use `<Trans />`

File: storybook/addons/SuspenseDecorator.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import React, { Suspense } from 'react';
+
+export default storyFn => <Suspense fallback={<div>loading</div>}>{storyFn()}</Suspense>;
```
Comment (Kamahl19) [inline storybook/addons/SuspenseDecorator.js]: what is this for exactly?

Comment (Kamahl19) [reply storybook/addons/SuspenseDecorator.js]: also you can use a fragment instead of div

File: storybook/config.js [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 import LocaleDecorator from './addons/LocaleDecorator';
-import Row from './addons/Row';
+import SuspenseDecorator from './addons/SuspenseDecorator';
 
 import '../src/app/styles/main.css';
 
+addDecorator(SuspenseDecorator);
 addDecorator(LocaleDecorator);
-addDecorator(Row.Decorator);
+addDecorator(StoryRouter());
```
Comment (Kamahl19) [inline storybook/config.js]: why dont we want the Row anymore?

Comment (MiroslavPetrik) [reply storybook/config.js]: We don't have the row in the common components.

We want to use other grid.

The naming is wrong, it should be just 'offset' or 'space' as we want to have margin arround the components so they are not sticked on the iframe window.

File: storybook/addons/SuspenseDecorator.js (hunk 3 above)
Comment (MiroslavPetrik) [reply storybook/addons/SuspenseDecorator.js]: > what is this for exactly?

I was getting suspense err when rendering the components, this fixed it. 💃 
Now I'll dig why there was the error.

Comment (MiroslavPetrik) [reply storybook/addons/SuspenseDecorator.js]:
```
Wrapper suspended while rendering, but no fallback UI was specified. Add a <Suspense fallback=...> component higher in the tree to provide a loading indicator or placeholder to display. in Wrapper (at story.js:6) in I18nextProvider (at LocaleDecorator.js:6) in InnerComponent (created by HistoryWatcher) in HistoryWatcher (created by Route) in Route (created by StoryRouter) in Router (created by MemoryRouter) in MemoryRouter (created by StoryRouter) in StoryRouter
```

Comment (MiroslavPetrik) [reply storybook/addons/SuspenseDecorator.js]: @Kamahl19 it's required for the `withTranslations` HoC.
I guess I will wrap just the `LocaleDecorator` with the Suspense then...

**Pull request merged**